horace_core/sqw/@sqw/parse_pixel_indices.m

```diff
@@ -1,212 +1,206 @@
-%   >> parse_pixel_indices (win,indx)
+%   >> [irun, idet, ien] = parse_pixel_indices (win)        % for all pixels
+%   >> [irun, idet, ien] = parse_pixel_indices (win, ipix)  % selected pixels
+%
+% Note: an efficient way of checking that ipix is consistent with all the sqw
+% objects is to call without output arguments i.e.
+%
+%   >> parse_pixel_indices (win, ipix)
+%
+% This will return silently without spending time computing the output
+% argument irun if the arguments are consistent, but will throw an error with an
+% appropriate message
 %
-% Get indices for a particular sqw object
-%   >> [irun,idet,ien] = parse_pixel_indices (win)           % all pixels
-%   >> [irun,idet,ien] = parse_pixel_indices (win,indx)      % pixels in indx (if scalar)
-%   >> [irun,idet,ien] = parse_pixel_indices (win,indx,iw)   % Index in indx if not scalar
 %
 % Input:
 % ------
 % Checking validity:
-%   win         Array of sqw objects, or cell array of scalar sqw objects
-%
-%   indx        Pixel indices to be extracted fromt he sqw object(s)
+%   win         Array of sqw objects
 %
-%               Single sqw object:
-%               ------------------
-%                 - ipix            Column vector of pixels indices
-%            *OR* - irun_idet_ien   Array of run, detector and energy bin index
-%                                  (array size nx3 where n is the number of pixels)
+% [Optional]
+%   ipix        Pixel indices for which the output is to be extracted from the
+%               sqw object(s)
 %
-%               Multiple sqw objects:
-%               ---------------------
-%                 - As above: assumed to apply to all sqw objects,
-%            *OR* - Cell array of the above, one cell array per sqw object
-%                  e.g. if two sqw objects:
-%                       {ipix1, ipix2}
-%                       {irun_idet_ien_1, irun_idet_ien_2}
+%               - Array of pixel indices. If there are multiple sqw objects,
+%                 it is then applied to every sqw object
 %
-% [Optional]
-%   iw          Indices of the sqw objects in win into the indexing arrays
-%              given by indx, one index for each for the sqw objects in win.
-%               This can be used to fine tune the use of indx for a specific
-%              sqw object or set of sqw objects without having to divide
-%              up indx. If indx has only one entry, then it is assumed to
-%              apply for any value of iw.
+%               - Cell array of pixel indices arrays
+%                   - only one array: applied to every sqw object, or
+%                   - several pixel indices arrays: one per sqw object
 %
 % Output:
 % -------
-%   irun        Single sqw object: Indexes of the experiments, containing
+%   irun        Single sqw object: Array of indices into the experiment_info
+%               Multiple sqw objects: Cell array of arrays, one per sqw object
+%                                  Cell array has the same size as win
```

Comment:
May be worth saying "shape" rather than "size", even though in MATLAB they're conflated.

Reply:
Yes, the conflation is annoying. But I'll leave as is because of the inherent Matlab conflation (if I'm not concentrating I'll occasionally type 'resize' when I  should be typing 'reshape', and 'shape' for the function called 'size')